tests/src/testfemixin.cpp

```diff
@@ -0,0 +1,107 @@
+#include <dune/functions/functionspacebases/powerbasis.hh>
+#include <dune/grid/yaspgrid.hh>
+
+using Dune::TestSuite;
+
+#include <ikarus/assembler/simpleassemblers.hh>
+#include <ikarus/finiteelements/fefactory.hh>
+#include <ikarus/finiteelements/mechanics/materials.hh>
+#include <ikarus/finiteelements/mechanics/materials/svk.hh>
+#include <ikarus/finiteelements/mechanics/nonlinearelastic.hh>
+#include <ikarus/utils/basis.hh>
+#include <ikarus/utils/init.hh>
+
+template <typename PreBasis>
+auto SimpleAssemblersTest(const PreBasis& preBasis) {
+  TestSuite t("SimpleAssemblersTest");
+  using Grid = Dune::YaspGrid<2>;
+
+  Dune::FieldVector<double, 2> bbox       = {4, 2};
+  std::array<int, 2> elementsPerDirection = {2, 1};
+  auto grid                               = std::make_shared<Grid>(bbox, elementsPerDirection);
+
+  auto gridView = grid->leafGridView();
+
+  auto basis        = Ikarus::makeBasis(gridView, preBasis);
+  auto matParameter = Ikarus::toLamesFirstParameterAndShearModulus({.emodul = 1000, .nu = 0.3});
+  auto totalDOFs    = basis.flat().size();
+
+  Ikarus::StVenantKirchhoff matSVK(matParameter);
+  auto reducedMat = planeStress(matSVK, 1e-8);
+  using namespace Ikarus;
+
+  auto vL = []([[maybe_unused]] const Dune::FieldVector<double, 2>& globalCoord, const double& lamb) {
+    Eigen::Vector2d fext;
+    fext.setZero();
+    fext[1] = 2 * lamb;
+    fext[0] = lamb;
+    return fext;
+  };
+
+  Dune::BitSetVector<1> neumannVertices(gridView.size(2), true);
+  BoundaryPatch neumannBoundary(gridView, neumannVertices);
+
+  auto nBLoad = []<typename VectorType>([[maybe_unused]] const VectorType& globalCoord, const auto& lamb) {
+    Eigen::Vector2d fext;
+    fext.setZero();
+    fext[0] = lamb / 40;
+    return fext;
+  };
+
+  auto preFE = makeFE(
+      basis, skills(nonLinearElastic(reducedMat), volumeLoad(vL), neumannBoundaryLoad(&neumannBoundary, nBLoad)));
+
+  std::vector<decltype(preFE())> fes;
+
+  for (auto&& ge : elements(gridView)) {
+    fes.emplace_back(preFE());
+    fes.back().bind(ge);
+  }
+
+  Eigen::VectorXd d(basis.flat().size());
+
+  d.setRandom();
+  double load = 0.0;
+
+  Ikarus::FERequirements req = Ikarus::FERequirements()
+                                   .insertGlobalSolution(Ikarus::FESolutions::displacement, d)
+                                   .insertParameter(Ikarus::FEParameter::loadfactor, load)
+                                   .addAffordance(Ikarus::AffordanceCollections::elastoStatics);
+
+  Eigen::VectorXd forces(fes.begin()->size());
+  forces.setZero();
+  Eigen::MatrixXd stiffness(fes.begin()->size(), fes.begin()->size());
+  stiffness.setZero();
+  const auto firstFE = *fes.begin();
+  calculateVector(firstFE, req, forces);
+  calculateMatrix(firstFE, req, stiffness);
+  std::cout << "Scalar" << calculateScalar(firstFE, req) << std::endl;
+  std::cout << "forces" << forces << std::endl;
+  std::cout << "stiffness" << stiffness << std::endl;
+
```

Comment:
Are we testing anything here?

Reply:
And depending on that the test name has to be modified?

Reply:
I refactored the test and also fixed it to be `COMPILE_ONLY`